org.civicrm.reqangular/src/common/services/api/contract.js

```diff
@@ -0,0 +1,66 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/modules/apis',
+  'common/services/api'
+], function (_, apis) {
+  'use strict';
+
+  apis.factory('api.contract', ['$log', 'api', function ($log, api) {
+    $log.debug('api.contract');
+
+    return api.extend({
+
+      /**
+       * Returns all contracts.
+       *
+       * It chains an additional call to the `getfulldetails` endpoint to also return
+       * full details on contracts
+       *
+       * @param  {Object} params  matches the api endpoint params (ex. contact_id)
+       * @return {Promise}
+       */
+      all: function (params) {
+        $log.debug('ContractAPI.all');
+
+        params = params || {};
+
+        params['api.HRJobContract.getfulldetails'] = {
+          'jobcontract_id': '$value.id',
+          'return': 'leave'
```

Comment:
As I mentioned [here](https://github.com/civicrm/civihr/pull/1942#discussion_r122711967), the `getFullDetails` API doesn't support specifying return fields and this will simply be ignored. It's probably better to remove it from here to avoid confusions.

Btw, shouldn't this have been caught by the tests?